Didi/remove r3 information (#768)
telerik/maui-docs: knowledge-base/chart-line-series-datapoint-markers.md

---

> title: Create Data Point Markers on Chart LineSeries

Reviewer: ```suggestion
title: Create Data Point Markers on Chart's Line Series
```

> description: How you can show point markers in LineSeries on data points in RadChart

Reviewer: ```suggestion
description: Learn how to show data point markers in the Line Series of the RadChart when using the native Chart for Android and iOS.
```

type: how-to
page_title: Show Data Points on CartesianChart LineSeries
slug: chart-line-series-datapoint-markers
position: 
tags: maui, chart, points, markers, line series, dornet maui
ticketid: 1612671
res_type: kb
---

## Environment

| Version | Product | Author | 
| --- | --- | ---- | 
| 6.6.0 | Telerik UI for .NET MAUI Chart |[Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova)| 

## Description

> Is there a way to turn on Data Point Markers for RadCartesianChart LineSeries? This can be really helpful in case there is only one result the chart doesn't show the value of the result.

Reviewer: ```suggestion
Is there a way to turn on Data Point Markers for the `RadCartesianChart` Line Series? I want to use this when there is only one result and the Chart doesn't show the value of this result.
```

## Solution

> You would need to use the native chart for Android and iOS in order to display data point markers on LineSeries of the CartesianChart. 

Reviewer: ```suggestion
To display data point markers on the Line Series of the Cartesian Chart, use the native Chart for Android and iOS.
```

    <telerik:RadCartesianChart.Series>
        <telerik:LineSeries ItemsSource="{Binding Data}"
                        CategoryBinding="Date"
                        ValueBinding="Value" />
    </telerik:RadCartesianChart.Series>
</telerik:RadCartesianChart>
```

**2.** Add the following namespace:

```XAML
xmlns:telerik="http://schemas.telerik.com/2022/xaml/maui"
```

**3.** Add sample model and view model:

```C#
public class TemporalData
{
    public DateTime Date { get; set; }
    public double Value { get; set; }
}

public class ViewModel
{
    public ObservableCollection<TemporalData> Data { get; set; }

    public ViewModel()
    {
        this.Data = GetDateTimeData(6);
    }

    private static ObservableCollection<TemporalData> GetDateTimeData(int itemsCount)
    {
        var startDate = new DateTime(2015, 03, 01);

        ObservableCollection<TemporalData> items = new ObservableCollection<TemporalData>();
        for (int i = 0; i < itemsCount; i++)
        {
            TemporalData data = new TemporalData();
            data.Date = startDate.AddDays(i);
            data.Value = Math.Sin(i);

            items.Add(data);
        }

        return items;
    }
}
```

> **4.** Subscribe to the `HandlerChanged` event and use the native chart for adding point markers on Android and iOS:

Reviewer: ```suggestion
**4.** Subscribe to the `HandlerChanged` event and use the native Chart to add point markers on Android and iOS:
```

#endif
    }
}
```

**5.** For Android, define a `PointRenderer` class inside the `Platforms/Android` folder:

```C#
public class PointRenderer : Java.Lang.Object, global::Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Pointrenderers.IChartDataPointRenderer
{
    private global::Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Categorical.LineSeries lineSeries;
    public PointRenderer(global::Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Categorical.LineSeries lineSeries)
    {
        this.lineSeries = lineSeries;
    }
    public void RenderPoint(global::Android.Graphics.Canvas canvas, global::Com.Telerik.Widget.Chart.Engine.DataPoints.DataPoint dataPoint)
    {
        global::Com.Telerik.Widget.Palettes.PaletteEntry entry = this.lineSeries.Palette.GetEntry(this.lineSeries, this.lineSeries.CollectionIndex);
        using (global::Android.Graphics.Paint paint = new global::Android.Graphics.Paint())
        {
            paint.Color = new global::Android.Graphics.Color(entry.Fill);
            canvas.DrawCircle((float)dataPoint.CenterX, (float)dataPoint.CenterY, 12, paint);
        }
    }
}
```

> **6.** And this is the result:

Reviewer: As this is the result, don't add it to the list.
```suggestion
And this is the result:
```
